### PR TITLE
Add tensorboard logger in TorchTrainingRunner

### LINKED ISSUE #159: Add tensorboard logger in TorchTrainingRunner

### PR DESCRIPTION
**Pull request by mromanelli9**
## Description

This PR allows to attach additional [PyTorch Ignite loggers](https://pytorch.org/ignite/v0.4.9/contrib/handlers.html) to the engines inside `TorchTrainingRunner`.

Usage example:
```python
from ignite.contrib.handlers.tensorboard_logger import *

# Create function to setup loggers and attach them to engines
def loggers_factory(trainer, train_evaluator, evaluator):
    # Create a logger
    tb_logger = TensorboardLogger(log_dir="experiments/tb_logs")

    # Attach the logger to the trainer engine
    tb_logger.attach(
        trainer,
        event_name=Events.ITERATION_COMPLETED,
        log_handler=GradsHistHandler(model)
    )

# Pass `logger_factory`  using `additional_loggers` parameter
runner = TorchTrainingRunner(
    model=resnet,
    dataset=(train_dataset, test_dataset),
    callbacks=[],
    options=options,
    loggers_factory=loggers_factory,
)
```

## List of changes
- new `additional_loggers` parameter for `TorchTrainingRunner`
- reduce boilerplate code inside `TorchTrainingRunner`

## Associated GitHub Issues
Closes #159

### CODE REVIEW

**Comment by rlebret** [comment]
I have used it in the CIFAR10 training example. It works.

File: src/mlmodule/torch/runners.py [hunk 1]
```diff
@@ -283,6 +289,8 @@ class TorchTrainingRunner(
         TorchTrainingOptions,
     ]
 ):
+    loggers_factory: Optional[Callable[[Engine, Engine, Engine], None]] = None
```
Comment (smassonnet) [inline src/mlmodule/torch/runners.py:292]: I would be in favor of keep the runners API static as much as possible.

Would there be a problem putting loggers_factory in the options rather that adding a new argument ?

Comment (mromanelli9) [reply src/mlmodule/torch/runners.py:292]: Shouldn't be a problem. I'll make the change.

Comment (mromanelli9) [reply src/mlmodule/torch/runners.py]: See https://github.com/LSIR/mlmodule/pull/161/commits/f2abde6d27c3097ae53ce069b6fb6a43f98e0b07

Comment (smassonnet) [reply src/mlmodule/torch/runners.py]: Ok perfect !

**Review by smassonnet — APPROVED** [review]

**Pull request merged**